Lesson19/didem/script.js

```diff
@@ -0,0 +1,176 @@
+  SHOPPING CART APPLICATION
+===========================================================
+In this project, you'll create a simple Shopping Cart to
+simulate adding items, removing items, calculating totals,
+and applying discounts.
+
+You'll practice:
+1. Classes and objects
+2. Encapsulation and abstraction
+3. Methods (functions inside a class)
+4. Arrays and basic array methods (push, filter, find)
+5. Conditional statements (if-else)
+
+Below is a step-by-step guide with comments explaining
+each part. You can test each step by running the code in
+Node.js or a browser console.
+*/
+
+/*
+-----------------------------------------------------------
+  STEP 1: Create the ShoppingCart Class
+-----------------------------------------------------------
+1. Define a `ShoppingCart` class.
+2. Add a constructor that initializes an empty private 
+   array `#items` to store the cart items.
+3. Add a `viewCart` method to display all items in the cart.
+*/
+
+/*
+-----------------------------------------------------------
+  STEP 2: Add Items to the Cart
+-----------------------------------------------------------
+1. Create an `addItem` method in the `ShoppingCart` class.
+2. The method should:
+   - Accept `name`, `price`, and `quantity` as parameters.
+   - Check if the item already exists in the cart.
+     - If it exists, increase the quantity.
+     - Otherwise, add the new item to the `#items` array.
+*/
+
+/*
+-----------------------------------------------------------
+  STEP 3: Remove Items from the Cart
+-----------------------------------------------------------
+1. Add a `removeItem` method to the `ShoppingCart` class.
+2. The method should:
+   - Accept the `name` of the item to remove.
+   - Remove the item from the `#items` array if it exists.
+*/
+
+class ShoppingCart {
+  #items;
+
+  constructor() {
+    this.#items = [];
+  }
+
+  viewCart() {
+    console.log('Viewing cart....');
+    for (const item of this.#items) {
+      console.log(item);
+    }
+  }
+
+  addItem(name, price, quantity) {
+    console.log('Adding new item to the cart...');
+    for (const item of this.#items) {
+      if (name === item.name) {
+        console.log('increase the quantity.');
+        item.quantity++;
+        return;
+      }
+    }
+
+    const id = new Date().getMilliseconds();
+    this.#items.push({ id, name, price, quantity });
+  }
+
+  // In real life, we usually use id to find elements or do any manipulations
+  removeItem(name) {
+    for (let i = 0; i < this.#items.length; i++) {
+      const currentElement = this.#items[i];
+      if (name.toLowerCase() === currentElement.name.toLowerCase()) {
+        this.#items.splice(i, 1);
+        return;
+      }
+    }
+  }
+getTotal() {
+    let totalAmount = 0;
+    // Sepetteki her öğe için döngü
+    for (const item of this.#items) {
+      // Toplam fiyata (fiyat * miktar) ekle
+      totalAmount += item.price.amount * item.quantity;
```

Comment:
It’s not specified anywhere that each item in the cart should have a `price` object with `amount` and `currency`. Your code only works if you manually add prices in that format. But if someone calls `addItem('Pencil', 5, 1)`, then `getTotal()` will break because `price.amount` does not exist. So in this project, you should treat `price` as a simple number and calculate the total using `price * quantity`.
```suggestion
      totalAmount += item.price * item.quantity;
```